src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/GenerateTSqlModelRequest.cs

```diff
@@ -0,0 +1,53 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+using Microsoft.SqlTools.Hosting.Protocol.Contracts;
+using Microsoft.SqlTools.ServiceLayer.Utility;
+using Microsoft.SqlServer.Dac.Model;
+
+namespace Microsoft.SqlTools.ServiceLayer.DacFx.Contracts
+{
+    /// <summary>
+    /// Parameters to generate a SQL model
+    /// </summary>
+    public class GenerateTSqlModelParams
+    {
+        /// <summary>
+        /// Project uri
+        /// </summary>
+        public string ProjectUri { get; set; }
+
+        /// <summary>
+        /// The version of Sql Server to target
+        /// </summary>
+        public string ModelTargetVersion { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Sql script file paths.
+        /// </summary>
+        public string[]? FilePaths { get; set; }
+    }
+
+    /// <summary>
+    /// Result for the GenerateTSqlModel Request.
+    /// </summary>
+    public class GenerateTSqlModelResult : ResultStatus
+    {
+        public TSqlModel Model { get; set; }
```

Comment:
Can this actually be sent back over the JSON RPC channel? That generally is only able to handle simple POCO-type classes. 

To avoid potential issues we should have our own DTO (Data Transfer Object) defined that we create from the actual TSqlModel when sending it over JSON RPC

Reply:
@Charles-Gagnon I've been thinking about this and I don't we need to return the model for the create call because we can access it through STS.

Reply:
My plan was to add another request to getObjects from an existing model (returning them over the JSON RPC channel)

Reply:
Yeah, I was trying to think if we wanted to do that. 

Pro:
Simplifies this and reduces chance of huge model causing perf issues

Con:
Just adds an extra call that needs to be done (and multiple calls if you need multiple types of objects)

What you suggested seems fine though, neither seem like the obvious choice and this avoids any potential problems with huge models.